Updates / Fixes for Blender 2.8 and 2.9

scorpion81 opened the pull request:
Hi @Pyroevil,
 
first a big thanks for your great molecular addon :smiling_face_with_three_hearts: 
 
@PavelBlend and @MarBasso and me have made some additional small fixes and updates in order to attempt to keep it running under recent blender and python versions. There are also other forks around, but i didnt read them in detail and hence they also may have other small fixes included, which we are not aware about.

Maybe you are interested to merge those updates (from my fork, for example) back into your main repo ? Because your repo is the only one where users can file issues. Those may be fixed in the fork(s) but perhaps the users have more trouble to find the fixed versions this way.

Not sure whether our fixes "broke" something unintentionally, though. But the addon loads properly and seems to function from a quick glance. 

Bonus: i have set up a github action, which will automatically build 3 zips under windows, mac and linux runners right here on github on each commit. Those zips are unfortunately zipped twice. But if you extract the "inner" zip, you can use it as release zip which is directly installable from within blender.

PavelBlend committed:
fix bugs: now works in a blender 2.79b

PavelBlend committed:
UI fix

PavelBlend committed:
Hide Molecular Panel

PavelBlend committed:
remove old realesee

PavelBlend committed:
remove old Molecular Core (with Python)

PavelBlend committed:
move sources dir in root folder

PavelBlend committed:
remove binary files

PavelBlend committed:
edit setup bat

PavelBlend committed:
rename 'setup' files to 'setup32'

PavelBlend committed:
remove ToDo.txt file

PavelBlend committed:
edit README.md

PavelBlend committed:
1.0.4

PavelBlend committed:
precision=6 for all parameters

PavelBlend committed:
deleted global variables

PavelBlend committed:
divided into modules

PavelBlend committed:
addd git ignore file

PavelBlend committed:
Linking Group and Activate Others Linking Options

PavelBlend committed:
#9 Groups as an integer value

PavelBlend committed:
refactor cmolcore.pyx: PEP8

PavelBlend committed:
clear export data

PavelBlend committed:
move descriptions text in module

PavelBlend committed:
ui: PEP8 refactor

PavelBlend committed:
operators: PEP8 refactor

PavelBlend committed:
ui: fix bug in particles number

PavelBlend committed:
ui: enable props when others links property

PavelBlend committed:
minor edits

PavelBlend committed:
minor edits

PavelBlend committed:
minor edits

PavelBlend committed:
move labels text in module

PavelBlend committed:
move props text in props_define function

PavelBlend committed:
rename cmolcore to core

PavelBlend committed:
free memory

PavelBlend committed:
fix bug in memory free

PavelBlend committed:
1.0.5

PavelBlend committed:
support python 3.7 (blender daily build)

PavelBlend committed:
support win 32 bit

PavelBlend committed:
make release script

PavelBlend committed:
1.0.6

PavelBlend committed:
support blender 2.80

PavelBlend committed:
1.0.7

PavelBlend committed:
fix errors in 2.80

scorpion81 committed:
adapt setup.py for mac (and linux)

Note, under mac it is not so trivial to use openmp, so omitting here for now.

scorpion81 committed:
Create make_release_linux_mac.py

Note, hardcoded for python 3.7 and 64bits for now, tested under mac only

scorpion81 committed:
Rename make_release.py to make_release_win.py

scorpion81 committed:
add platform.system() == "Linux" check 

just in case platform.architecture()[1] is not "ELF" for some reason

scorpion81 committed:
fix, use "expected" linux .so name now

needs to have x86_64-linux-gnu in its name

scorpion81 committed:
molecular should work again now in 2.8

scorpion81 committed:
attempt to automate windows build, too

scorpion81 committed:
use sys.executable for subprocess

scorpion81 committed:
fix for linux build script, and added author

scorpion81 committed:
fix to reset particle cache prior to resimulation

scorpion81 committed:
minor fixes

for missing context.scene.update and set active uv operators as well as for Bake UV after ending

scorpion81 committed:
version bump due to a few fixes

scorpion81 committed:
fixed baked UV rendering

forced the angular velocities to be written into the pointcache as well

PavelBlend committed:
refactor

PavelBlend committed:
refactor make release script

PavelBlend committed:
refactor ui

PavelBlend committed:
fix tools panel name

PavelBlend committed:
total time info

scorpion81 committed:
adapt release file for linux/mac and version is 1.11 now

PavelBlend committed:
fix uv baking

scorpion81 committed:
unified release script

scorpion81 committed:
removed hardcoded 37m, use major minor abiflags instead

Commit:
Update time library for Python 3.3+

Updated the code to be compliant with Python 3.3+ distributions, in which the time.clock module is deprecated and replaced by time.perf_counter. To minimize the modifications, I imported the perf_counter module and locally named it as clock. With this approach, the rest of the code is untouched.

Commit:
Update time library for Python 3.3+

Updated the code to be compliant with Python 3.3+ distributions, in which the time.clock module is deprecated and replaced by time.perf_counter. To minimize the modifications, I imported the perf_counter module and locally named it as clock. With this approach, the rest of the code is untouched.

scorpion81 committed:
Create molecular-build.yml

scorpion81 committed:
create 3 building jobs, one for each os

scorpion81 committed:
trying to get the build job working

scorpion81 committed:
fix attempt for build system

scorpion81 committed:
add language level 3 for cython

scorpion81 committed:
hopefully this fixes the language level warning

scorpion81 committed:
fix for build error in windows with python 3.8.x

according to the documentation sys.abiflags shall return an empty string, but it is not even defined in under windows.

scorpion81 committed:
bump version of core.pyx

scorpion81 committed:
bump version of addon to 1.1.2

scorpion81 committed:
attempt to get correct version for packaging

scorpion81 committed:
how does this work....

scorpion81 committed:
another try

scorpion81 committed:
another fix attempt for getting a version

scorpion81 committed:
string tinkering...

scorpion81 committed:
another approach to get the version

scorpion81 committed:
hrmpf...

scorpion81 committed:
and another try...

scorpion81 committed:
broken pipe ? wtf...

scorpion81 committed:
test

scorpion81 committed:
fix for file name glitch (hopefully)

scorpion81 committed:
damn parenthesis...

scorpion81 committed:
update artifact creation rules for mac and linux

scorpion81 committed:
fix for loading error (undefined symbols)

errors with undefined functions from math.h may occur on loading the linux version of molecular.
So hopefully '-fno-builtin' compiler switch will fix this.

scorpion81 committed:
change linux runner to ubuntu 18.04

scorpion81 committed:
try to additionally link math lib (lm)

scorpion81 committed:
put status badge into readme

scorpion81 committed:
hrmpf, blender 2.90.1 still uses python 3.7

was assuming the switch to 3.8 was already made.
Trying to adapt this now in the builders.

bugcatching committed:
fix english grammar

PavelBlend commented:
Hello @scorpion81 

You can enable issues in your repository:
![01](https://user-images.githubusercontent.com/7983249/98436862-cd62aa00-20ef-11eb-9040-be3d424edead.jpg)
![02](https://user-images.githubusercontent.com/7983249/98436865-d05d9a80-20ef-11eb-8931-5d5c225b5e52.jpg)

scorpion81 committed:
linux: attempt to statically link against python

for some weird reason, there is no dynamic python library included in blenders linux release build
So in order to run molecular there, try to link statically against the according python version which should be included in blender.

scorpion81 committed:
Revert "linux: attempt to statically link against python"

This reverts commit 5f931c76af1d5c55541b8d7a1618c6dd14851a73.

scorpion81 committed:
compile against python 3.9 for blender 2.93+

update python version against which this addon is being compiled

scorpion81 committed:
attempt installing libpython.a for linux

try calling into apt get with a custom ppa in order to setup libpython.a / libpython3-devel package

scorpion81 committed:
fix: forgot sudo

scorpion81 committed:
fix: corrected package name

scorpion81 committed:
update used python version to 3.10

scorpion81 committed:
fix syntax error with python 3.10

scorpion81 committed:
version bump

scorpion81 committed:
build a "fat" binary for mac os

scorpion81 committed:
removed dead links

scorpion81 committed:
Hmm, M1 seems to be arm64e, not arm64

scorpion81 committed:
attempt to automate draft releases

scorpion81 committed:
action yml syntax error fixes

scorpion81 committed:
fix workflow logic

scorpion81 committed:
yet another fix

scorpion81 committed:
hrm, cache doesnt work here, using artifacts again

scorpion81 committed:
try to upload this now

scorpion81 committed:
gah, forgot to checkout the repo

scorpion81 committed:
fixes for type error and crash and version bump

scorpion81 committed:
added some UV Baking information to the readme

Addes some short explanation and example blend file on how to add UVs to the particle system instances.

scorpion81 committed:
some clarification attempt 

attempt to clarify the proper usage of the particle info node

scorpion81 committed:
update fat binary for arm64

scorpion81 committed:
added some readme how to build manually

scorpion81 committed:
update/recompilation to Python 3.11, used in Blender 4.1

scorpion81 committed:
build linux version on ubuntu 22.04 runner

scorpion81 committed:
fix to adapt to changes in blender API related to contexts

scorpion81 committed:
Update core.pyx

revert pow operator usage to old behavior  with cpow(True)
otherwise there may be errors

scorpion81 committed:
Update __init__.py

version bump

scorpion81 committed:
Update molecular-build.yml

replace set-output directives by using GITHUB_OUTPUT environment files, because set-output is about to be deprecated

scorpion81 committed:
Update core.pyx

revert cpow changes, compile error on windows

scorpion81 committed:
Update setup.py

attempt to set cpow to True via compiler directive

scorpion81 committed:
Update molecular-build.yml

attempt to fix build file using v4 artifact actions and proper usage of outputs.version

scorpion81 committed:
Update setup.py

compiler_directives is unknown , try cython_directives

scorpion81 committed:
Update molecular-build.yml

attempt to use bash shell syntax on windows runner as well

scorpion81 committed:
Update __init__.py

bump to correct version
